infra/base-images/base-builder/compile_gfuzztests.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash -eu
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+bazel run @com_google_fuzztest//bazel:setup_configs > fuzztest.bazelrc
+
+# Bazel target names of the fuzz binaries
+FUZZ_TEST_BINARIES=$(bazel query 'kind("cc_test", rdeps(..., @com_google_fuzztest//fuzztest:fuzztest_gtest_main))')
+
+# Bazel output paths of the fuzz binaries
+FUZZ_TEST_BINARIES_OUT_PATHS=$(bazel cquery 'kind("cc_test", rdeps(..., @com_google_fuzztest//fuzztest:fuzztest_gtest_main))' --output=files)
+
+#echo "Fuzz test binaries:"
+#echo ${FUZZ_TEST_BINARIES}
+#echo "-----------------------------"
+#echo ${FUZZ_TEST_BINARIES3}
+#echo "-----------------------------"
+
+# Build the project and fuzz binaries
+bazel build --subcommands --config=oss-fuzz ${FUZZ_TEST_BINARIES[*]}
+
+# Iterate the fuzz binaries and list each fuzz entrypoint in the binary. For
+# each entrypoint create a wrapper script that calls into the binaries the
+# gien entrypoint as argument.
+# The scripts will be named:
+# {binary_name}_GFUZZTEST_{fuzztest_entrypoint}
+for fuzz_main_file in $FUZZ_TEST_BINARIES_OUT_PATHS; do
+  FUZZ_TESTS=$($fuzz_main_file --list_fuzz_tests)
+  cp ${fuzz_main_file} $OUT/
+  fuzz_basename=$(basename $fuzz_main_file)
+  for fuzz_entrypoint in $FUZZ_TESTS; do
+    TARGET_FUZZER="${fuzz_basename}_GFUZZTEST_$fuzz_entrypoint"
+
+    # Write executer script
+    echo "#!/bin/sh
+  # LLVMFuzzerTestOneInput for fuzzer detection.
```

Comment:
I don't understand this LLVMFuzzerTestOneInput comment.

Reply:
This is specific to OSS-Fuzz as OSS-Fuzz identifies targets by searching for "LLVMFuzzerTestOneInput"  in the target file when checking for bad builds in the CI. The specific line that does it is here: https://github.com/google/oss-fuzz/blob/3682b658055f9b7f2fbe43ea38a1a3892c322894/infra/base-images/base-runner/test_all.py#L95